packages/swing-store/src/snapStore.js

```diff
@@ -12,20 +12,29 @@ import {
 } from '@agoric/internal';
 
 /**
- * @typedef {object} SnapshotInfo
+ * @typedef {object} SnapshotResult
  * @property {string} hash sha256 hash of (uncompressed) snapshot
  * @property {number} rawByteCount size of (uncompressed) snapshot
  * @property {number} rawSaveSeconds time to save (uncompressed) snapshot
  * @property {number} compressedByteCount size of (compressed) snapshot
  * @property {number} compressSeconds time to compress and save snapshot
  */
 
+/**
+ * @typedef {object} SnapshotInfo
+ * @property {number} endPos
+ * @property {string} hash
+ */
+
 /**
  * @typedef {{
  *   has: (hash: string) => boolean,
- *   load: <T>(hash: string, loadRaw: (filePath: string) => Promise<T>) => Promise<T>,
- *   save: (saveRaw: (filePath: string) => Promise<void>) => Promise<SnapshotInfo>,
- *   deleteSnapshot: (hash: string) => void,
+ *   load: <T>(vatID: string, loadRaw: (filePath: string) => Promise<T>) => Promise<T>,
+ *   save: (vatID: string, endPos: number, saveRaw: (filePath: string) => Promise<void>) => Promise<SnapshotResult>,
+ *   deleteVatSnapshots: (vatID: string) => void,
+ *   deleteUnusedSnapshots: () => void,
+ *   deleteSnapshotByHash: (hash: string, vatID: string) => void,
+ *   getSnapshotInfo: (vatID: string) => SnapshotInfo,
```

Comment:
`has` is used in tests so it needs to stay in, though a case can be made that it should acquire a vatID parameter.

In a similar vein, `deleteUnusedSnapshots` and `deleteSnapshotByHash` are currently only used by tests, though one can imagine future maintenance or debugging tools that might use them.  I'm very disinclined to fall back on using the sqlite command line client to do things like clear out unused snapshots, because that implies a level of understanding of the schema and a level of care using the sqlite tool that I wouldn't want to count on operators (or even myself after stepping away from this code for a few weeks) to necessarily have.

I'm a bit wishy-washy about getting rid of the word "snapshot" in the method names, given folks' propensity to use destructuring on API objects, but your comment does point out a naming inconsistency in that some method names have it and some don't -- if we want to have the word "snapshot" in the names, then for example `save` should probably be `saveSnapshot`.  And putting "snapshot" in all of them definitely seems silly.  So, yeah, we can clean that up.  However, all of the non-maintenance, non-test operations (i.e., the ones that are used normally and ubiquitously) are per-vat and operate on the latest snapshot, so by the same reasoning I don't think putting "vat" or "latest" in the names helps.

It also strikes me that it's probably good to be consistent about parameter order -- the large majority of methods that take a vatID parameter should have it in the same (first) position, which is not currently universal.

So I think the main API ends up as:
 - `load(vatID, loadRaw)` (unchanged)
 - `save(vatID, endPos, saveRaw)`  (unchanged)
 - `getInfo(vatID)` (was `getSnapshotInfo`)
 - `deleteAll(vatID)` (was `deleteVatSnapshots`)
 
with testing/debug/maintenance methods:
 - `hasHash(vatID, hash)` (was `has(hash)`)
 - `deleteByHash(vatID, hash)` (was `deleteSnapshotByHash(hash, vatID)`)
 - `deleteAllUnused()` (was `deleteUnusedSnapshots`)

How does that all work for you?

Reply:
After making the above changes to see what it looked/felt like in the code, I'm reconsidering the names.  Part of the problem is that the API object is called "snapStore" and not "snapshotStore".  Perhaps we might want to change _that_, but I'm leaning towards mostly keeping the old/existing names. I've found that naked `getInfo` et al looks strange in the actual use context.  Now I'm leaning towards:
 - `loadSnapshot(vatID, loadRaw)` (was `load`)
 - `saveSnapshot(vatID, endPos, saveRaw)` (was `save`)
 - `getSnapshotInfo(vatID)` (now unchanged)
 - `deleteSnapshots(vatID)` (now unchanged

and then:
- `hasHash(vatID, hash)` (was `hasHash(hash)` -- keep the added vatID parameter from the above proposal)
- `deleteSnapshotByHash(vatID, has)` (was same name but make the parameter order consistent)
- `deleteUnusedSnapshots` (now unchanged)

Now what do you think?

Reply:
looks good, although `deleteUnusedSnapshots` could be `deleteAllUnusedSnapshots` to highlight the fact that it doesn't take a `vatID` argument